text/src/sort.rs

```diff
@@ -0,0 +1,887 @@
+
+    let start_2 = parts.next();
+    let mut start_2: usize = match first {
+        true => start_2
+            .unwrap_or("1")
+            .parse()
+            .map_err(|err| Box::new(Error::new(ErrorKind::Other, err)))?,
+        false => start_2
+            .unwrap_or(&usize::MAX.to_string())
+            .parse()
+            .map_err(|err| Box::new(Error::new(ErrorKind::Other, err)))?,
+    };
+
+    if !first && start_2 == 0 {
+        start_2 = usize::MAX;
+    }
+
+    Ok(RangeField {
+        field_number: start_1 - 1,
+        first_character: start_2 - 1,
+        numeric_sort,
+        ignore_leading_blanks,
+        reverse,
+        ignore_nonprintable,
+        fold_case,
+        dictionary_order,
+    })
+}
+
+/// Compares two strings based on a specified key range and optional field separator.
+///
+/// This function compares two strings (`line1` and `line2`) based on a specified key range,
+/// taking into account optional field separation if provided. It applies sorting and comparison
+/// rules defined in the `RangeField` struct associated with the key range.
+///
+/// # Arguments
+///
+/// * `line1` - A string slice (`&str`) representing the first input string to be compared.
+/// * `line2` - A string slice (`&str`) representing the second input string to be compared.
+/// * `key_range` - A tuple containing two elements:
+///     * The first `RangeField` specifies the key range configuration for both strings.
+///     * An optional `RangeField` specifies the end of the key range if different from the start.
+/// * `field_separator` - An optional character specifying the field separator for splitting
+///   strings into fields before comparison. If `None`, whitespace is used as the separator.
+///
+/// # Returns
+///
+/// An `Ordering` value (`Ordering::Less`, `Ordering::Greater`, or `Ordering::Equal`) indicating
+/// the result of the comparison:
+/// * `Ordering::Less` if `line1` is less than `line2` according to the specified key range.
+/// * `Ordering::Greater` if `line1` is greater than `line2` according to the specified key range.
+/// * `Ordering::Equal` if `line1` and `line2` are equal within the specified key range.
+///
+fn compare_key(
+    line1: &str,
+    line2: &str,
+    key_range: &(RangeField, Option<RangeField>),
+    field_separator: Option<char>,
+) -> Ordering {
+    let mut line1 = {
+        if let Some(separator) = field_separator {
+            let split = line1.split(separator);
+            if key_range.0.ignore_leading_blanks {
+                cut_line_by_range(split.skip_while(|s| s.is_empty()).collect(), key_range)
+            } else {
+                cut_line_by_range(split.collect(), key_range)
+            }
+        } else if key_range.0.ignore_leading_blanks {
+            cut_line_by_range(line1.split_whitespace().collect(), key_range)
+        } else {
+            cut_line_by_range(line1.split(' ').collect(), key_range)
```

Comment:
Remove duplication. Make `if` block return the vec and call `cut_line_by_range` with the result